### ISSUE TITLE
Is there a way to automatically generate prerequisites (imports) for a jsonnet file for constructing a Makefile rule?

### ISSUE THREAD

**Issue by varunbpatil** [google/jsonnet#833]
The C compiler, for example, has a way to [automatically generate all the prerequisites](https://www.gnu.org/software/make/manual/html_node/Automatic-Prerequisites.html#Automatic-Prerequisites) (.h header files) for a given source (.c) file.

Similarly, is there a way to get all the dependencies (imports) of a jsonnet file?

**Comment by sbarzowski** [comment]
We don't have a tool ready, but it should be quite easy to build with API in https://github.com/google/go-jsonnet. We even have some recursive traversal of imports implemented – it's enough to extract that and dump them.

One issue is that the normalized paths are absolute and that's probably not ideal for the Makefile use case, but we can just clip the prefix.

**Comment by varunbpatil** [comment]
Thanks @sbarzowski , will take a look at the Go API's.

**Comment by sbarzowski** [comment]
This is the most interesting function https://github.com/google/go-jsonnet/blob/v0.16.0/vm.go#L248 – it imports a program and gives you a syntax tree. You can then traverse it easily with a recursive function using `toolutils.Children` helper: https://github.com/google/go-jsonnet/blob/v0.16.0/toolutils/ast.go#L10. Here's an example: https://github.com/google/go-jsonnet/pull/325/files#diff-80f9fdcc81aed508ba78619f9dd78083R111.

We are interested in providing this functionality as an official tool. I'll be happy to help with an effort to build it.

**Comment by varunbpatil** [comment]
Thanks for those pointers @sbarzowski. I'm not an expert in Go, but, I was able to build a simple solution using the functions you pointed to. I'd be happy to contribute.

My solution doesn't provide a full stack trace like the code in interpreter.go when a file is not present. I will try to explore how to get the full stack trace.

https://github.com/varunbpatil/go-jsonnet/commit/75cccc10dd2c2ae0860bec3dfebec65575d680ca
I added a new parameter -d/--dependencies to jsonnet cli to list the dependencies.
